zipline/data/bundles/alpha_vantage_api.py

```diff
@@ -0,0 +1,239 @@
+
+
+@bundles.register('alpha_vantage', calendar_name="NYSE", minutes_per_day=390)
+def api_to_bundle(interval=['1m']):
+    def ingest(environ,
+               asset_db_writer,
+               minute_bar_writer,
+               daily_bar_writer,
+               adjustment_writer,
+               calendar,
+               start_session,
+               end_session,
+               cache,
+               show_progress,
+               output_dir
+               ):
+
+        divs_splits = {'divs': pd.DataFrame(columns=['sid', 'amount',
+                                                  'ex_date', 'record_date',
+                                                  'declared_date', 'pay_date']),
+                       'splits': pd.DataFrame(columns=['sid', 'ratio',
+                                                    'effective_date'])}
+
+        def minute_data_generator():
+            return (sid_df for (sid_df, *metadata.iloc[sid_df[0]]) in 
+                df_generator(interval='1m', start=start_session, end=end_session, divs_splits=divs_splits))
+
+        def daily_data_generator():
+            return (sid_df for (sid_df, *metadata.iloc[sid_df[0]]) 
+                in df_generator(interval='1d', start=start_session, end=end_session, divs_splits=divs_splits))
+
+        metadata = metadata_df()
+
+        for _interval in interval:
+            if _interval == '1d':
+                daily_bar_writer.write(daily_data_generator(), show_progress=True)
+            elif _interval == '1m':
+                minute_bar_writer.write(minute_data_generator(), show_progress=True)
+
+        # convert back wrong datatypes after pd.concat
+        divs_splits['splits']['sid'] = divs_splits['splits']['sid'].apply(lambda x: np.int(x))
+        divs_splits['divs']['sid'] = divs_splits['divs']['sid'].apply(lambda x: np.int(x))
+        divs_splits['divs']['ex_date'] = divs_splits['divs']['ex_date'].apply(lambda x: pd.to_datetime(x))
+        divs_splits['divs']['record_date'] = pd.to_datetime(divs_splits['divs']['record_date'])
+        divs_splits['divs']['declared_date'] = pd.to_datetime(divs_splits['divs']['declared_date'])
+        divs_splits['divs']['pay_date'] = pd.to_datetime(divs_splits['divs']['pay_date'])
+        
+        adjustment_writer.write(splits=divs_splits['splits'], dividends=divs_splits['divs'])
+
+        # Drop the ticker rows which have missing sessions in their data sets
+        metadata.dropna(inplace=True)
+
+        asset_db_writer.write(equities=metadata)
+        print(metadata)
+
+    return ingest
+
+
+if __name__ == '__main__':
+    from zipline.data.bundles import register
+    from zipline.data import bundles as bundles_module
+    import trading_calendars
+    import os
+    import time
+
+    cal: TradingCalendar = trading_calendars.get_calendar('NYSE')
+
+    # alpha-vantage has a fixed time-window, no point in changing these
+    end_date = pd.Timestamp('now', tz='utc').date() - timedelta(days=1)
```

Comment:
what if we do it today after market hours? will we miss today's data?

Reply:
yes, data is updated next day as far as i can see